### PR TITLE
FLEX-291: Remove lodash as a dependency

### PR DESCRIPTION
**Pull request by mjsalinger**

### CODE REVIEW

File: lib/service/modules/kinveyEntity.js [hunk 1]
```diff
@@ -50,98 +48,98 @@ function kinveyEntityModule(environmentId, useBSONObjectId) {
...
+        if (!_entity._kmd.ect) _entity._kmd.ect = currentTimeString;
+        if (!_entity._kmd.lmt) _entity._kmd.lmt = currentTimeString;
       }
     }
 
     let aclFunctions = {};
 
     function getCreator() {
-      return getProp(_entity, ['_acl', 'creator']);
+      return _entity._acl.creator;
```
Comment (samlevin) [inline lib/service/modules/kinveyEntity.js]: Will throw a `TypeError` if `_entity._acl` is null. Applies to some other occurrences below

File: lib/service/moduleGenerator.js [hunk 2]
```diff
@@ -101,15 +98,12 @@ function generateModules(task) {
...
-    appMetadata
-  );
+  requestMetadata.securityContext = getSecurityContextString(task.request.headers.authorization, appMetadata);
 
   let useBSONObjectId = false;
-  const migrated = getProp(task, ['appMetadata', 'appMetadata', 'objectid_migration', 'status']);
-  if (migrated != null && migrated === 'done') {
-    useBSONObjectId = true;
+
+  if (task.appMetadata.maintenance && task.appMetadata.maintenance.objectid_migration) {
```
Comment (samlevin) [inline lib/service/moduleGenerator.js]: The path on this line has changed. Was the original path a bug? `getProp(task, ['appMetadata', 'appMetadata', 'objectid_migration', 'status'])`

Comment (mjsalinger) [reply lib/service/moduleGenerator.js]: Yes, it was a bug.

File: lib/service/modules/kinveyEntity.js (hunk 1 above)
Comment (mjsalinger) [reply lib/service/modules/kinveyEntity.js]: No, it won't, because of line 53 above.

Comment (samlevin) [reply lib/service/modules/kinveyEntity.js]: Missed that 👍

**Pull request merged**